src/plugins/data_source_management/public/types.ts

```diff
@@ -39,11 +39,20 @@ export interface DataSourceManagementContext {
   authenticationMethodRegistry: AuthenticationMethodRegistry;
 }
 
+export enum DataSourceConnectionType {
+  OpenSearchConnection,
+  DirectQueryConnection,
+}
+
 export interface DataSourceTableItem {
   id: string;
+  type: string | undefined;
```

Comment:
Any reason for not doing
```suggestion
  type?: string;
```